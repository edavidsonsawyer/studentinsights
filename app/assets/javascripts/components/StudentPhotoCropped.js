import React from 'react';
import PropTypes from 'prop-types';
import * as Routes from '../helpers/Routes';
import {enhancedStudentPhotoStyles} from '../helpers/PerDistrict';


// Pure UI component for displaying a student photo, cropped to be
// smaller and zoomed-in to student's face.
// See also <StudentPhoto />
export default class StudentPhotoCropped extends React.Component {
  render() {
    const {districtKey} = this.context;
    const {studentId, style = {}} = this.props;
    const perDistrictStyles = enhancedStudentPhotoStyles(districtKey);
    return (
      <div
        className="StudentPhotoCropped"
        style={{
          width: 58,
          height: 58,
          marginRight: 10,
          border: '1px solid #aaa',
          borderRadius: 3,
          backgroundColor: '#ccc',
          backgroundImage: `url(${Routes.studentPhoto(studentId)})`,
          overflow: 'hidden',
          backgroundPosition: 'center top',
          backgroundRepeat: 'no-repeat',
          ...perDistrictStyles,
          ...style
        }}></div>
    );
  }
}
StudentPhotoCropped.contextTypes = {
  districtKey: PropTypes.string.isRequired
};
StudentPhotoCropped.propTypes = {
  studentId: PropTypes.number.isRequired,
  style: PropTypes.object,
};
