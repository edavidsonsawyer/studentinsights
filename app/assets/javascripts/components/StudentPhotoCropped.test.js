import React from 'react';
import ReactDOM from 'react-dom';
import StudentPhotoCropped from './StudentPhotoCropped';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const el = document.createElement('div');
  ReactDOM.render(<StudentPhotoCropped studentId={42} />, el);
});

it('snapshots', () => {
  const tree = renderer
    .create(<StudentPhotoCropped studentId={42} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
