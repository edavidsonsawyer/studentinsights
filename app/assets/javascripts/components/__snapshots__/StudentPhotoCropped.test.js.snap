// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#ccc",
      "backgroundImage": "url(/students/42/photo)",
      "backgroundPosition": "center top",
      "backgroundPositionY": -10,
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "150%",
      "border": "1px solid #aaa",
      "borderRadius": 3,
      "height": 58,
      "marginRight": 10,
      "overflow": "hidden",
      "width": 58,
    }
  }
/>
`;
